applications/nrf_desktop/doc/dfu.rst

```diff
@@ -0,0 +1,129 @@
+
+* Version and length of the image.
+* Partition ID of the image to specify its placement on the flash.
+
+reboot
+------
+
+This command is issued by sending a ``config_fetch_event`` from the host tool through the DFU module, with the ``reboot`` option used for identification.
+When received, it triggers an instant reboot of the device.
+
+start
+-----
+
+The ``start`` command is issued by sending a ``config_event`` from the host tool through the DFU module, with the ``start`` option used for identification.
+The command contains the following data:
+
+* Update image checksum.
+* Size of the image.
+* Offset at which the host tool would like to start an image update.
+  If the image transferred is performed for the first time, the offset must be set to zero.
+
+When the command is issued, the checksum and the size are recorded and the update process will be started.
+
+If for some reason the transmission is interrupted, the current offset position is stored along with the image checksum and size until the device is rebooted.
+The host tool can restart the update image transfer after interruption.
+In such case, the checksum and the size of the image (as well as the requested offset) must match the information stored by the device.
+
+data
+----
+
+This command is issued by sending a ``config_event`` from the host tool through the DFU module, with the ``data`` option used for identification.
+The command passes the chunk of the update image that should be stored at the current offset.
+
+If the operation is successful, the offset is increased.
+In case of failure the update process is interrupted.
+
+For performance reasons, this command does not return any information back to the host.
+To check that the update process is correct, the host tool must issue a ``sync`` command in regular intervals.
+
+.. note::
+    The DFU module does not check if the update image contains any valid data.
+    It also does not check if it is correctly signed.
+    When the update image is received, the host tool should request a reboot.
+    Then the bootloader will check the image for validity and ensure the signature is correct.
+    If verification of the new images is successful, the new version of the application will boot.
+
+sync
+----
+
+The ``sync`` command is issued by sending a ``config_fetch_event`` from the host tool with the DFU module and the ``sync`` option used for identification.
+The command confirms to the tool that the update process is ongoing.
+It also sends back the following data:
+
+* Checksum.
+* Size of the image transmitted so far.
+* Offset at which the update process currently is.
+
+The update tool can issue the ``sync`` command before starting the update process to see at which offset the update is to be restarted.
+
+Partition preparation
+=====================
+
+The DFU module must prepare the partition before the update image can be stored.
+This operation is done in the background.
+
```

Comment:
I think we should mention that DFU cannot be started until the memory is clean.
The user should not use the device in that case and wait until the operation will be finished.

Reply:
Fixed